### DISCUSSION TITLE
Run Bukkit/Spigot/Paper projects directly

### DISCUSSION THREAD

**Discussion by Lasslos** [minecraft-dev/MinecraftDev#1315] [Ideas]
Hey Guys!

I recently taught about adding an option to run your written plugins directly.

How would that work?

1. Download BuildTools (Bukkit, Spigot) or PaperMC
2. Run `java -jar "BuildTools.jar" --rev <Minecraft-Version> -o /Server`
3. Create start.bat
4. Ask the user if he agrees to Minecraft EULA
5. Compile the plugin
6. Copy the output to /Server/plugins
7. Run start.bat

All of that - but the end user only has to press one button.

That's it! As you can see, its not that much to implement. I would do it, I just wanted to know if you would support this.

**Comment by MrAn0nym** [comment]
you should probably start the server directly via java and not with the batch file to prevent problems with Linux and MacOS

**Reply by MarcPG1905** [reply]
You don't even have to use batch files, you can just use `java -jar server.jar`, which is the same command on every operating system.

**Comment by Earthcomputer** [comment]
In terms of the eula, it might be better not to mess with that. Just let the user agree to the eula in the normal way when they start a server.

Also, instead of copying the plugin, it should be a symlink where possible.

**Comment by DenWav** [comment]
Someone mentioned setting up server connections like services with database connections and Docker containers. That would be a cool feature potentially, but would be non-trivial to build.

**Comment by MarcPG1905** [comment]
This way would be easier:

1. Download the server files (BuildTools / Paper Server JAR)
2. Do the server setup (depends if it's BuildTools or Paper)
3. Make the user agree to the EULA
4. Compile the plugin into the server's plugin folder
5. Run `java -jar server.jar`

**Reply by Lasslos** [reply]
Yes, you're right. Are you looking to implement such a feature? I haven't done any work in it recently but I might pick it up again if there is interest.

**Reply by MarcPG1905** [reply]
I might start working on something like this today, but I have some other projects to finish and then I would have to wait until the pull request gets approved.
